### PR TITLE
feat(app): add network change to robot settings

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add wifi connection details screen and NetworkDetailsModal

Screenshot 2023-04-27 at 10 44 07 AM

In terms of WiFi list, it requires to update and will be updated by a following PR. There is WifiConnectionDetails under `SetupNetwork`. I'd like to keep this for a while to utilize for refactoring.

Screenshot 2023-04-27 at 10 44 20 AM

# Test Plan

# Changelog
- add WifiConnectionDetails + test
- add NetworkDetailsModal + test
- update NetworkSettings test
- add info icon

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/NetworkDetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      modalSize="medium"
+      header={modalHeader}
+      onOutsideClick={() => setShowNetworkDetailModal(false)}
+    >
+      <Flex
+        flexDirection={DIRECTION_COLUMN}
+        gridGap={SPACING.spacing3}
+        width="100%"
+      >
+        {/* ip address */}
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/NetworkDetailsModal.tsx]: do these comments add context when both the name and the value props are the same combination of words?

Comment (koji) [reply app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/NetworkDetailsModal.tsx]: yes
Should I remove them?

File: app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/__tests__/NetworkDetailsModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should render text and icon - wifi', () => {
+    const [{ getByText, getByLabelText }] = render(props)
+    // aria-label={`icon_${iconName}`}
```
Comment (jerader) [inline app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/__tests__/NetworkDetailsModal.test.tsx]: forgotten comment

File: app/src/organisms/OnDeviceDisplay/SetupNetwork/index.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 export * from './ConnectingNetwork'
 export * from './DisplayWifiList'
-export * from './EthernetConnectionDetails'
+export * from '../RobotSettingsDashboard/NetworkSettings/EthernetConnectionDetails'
```
Comment (jerader) [inline app/src/organisms/OnDeviceDisplay/SetupNetwork/index.ts]: this export is a bit confusing to me. You added it here so you can import `EthernetConnectionDetails` from `SetupNetwork`?

Comment (koji) [reply app/src/organisms/OnDeviceDisplay/SetupNetwork/index.ts]: oh, good catch. thank you.

File: components/src/icons/icon-data.ts [hunk 4]
```diff
@@ -670,4 +670,9 @@ export const ICON_DATA_BY_NAME = {
     path:
       'M14.4004 3.59998H33.6004V27.6H30V36H18V27.6H14.4004V3.59998ZM26.3996 43.2V36H21.5996V43.2H26.3996Z',
   },
+  info: {
```
Comment (ewagoner) [inline components/src/icons/icon-data.ts]: This looks like the old un-alphabetized file.

Comment (koji) [reply components/src/icons/icon-data.ts]: merged edge into this branch, so that should be solved.

Comment (ewagoner) [reply components/src/icons/icon-data.ts]: Yours is still at the bottom though. Can you slide it up?

Comment (koji) [reply components/src/icons/icon-data.ts]: oh thanks, I will slide `info` up.

**Review by ewagoner — APPROVED** [review]

**Review by jerader — APPROVED** [review]
lgtm!

**Pull request merged**